tests/Feature/Api/V1Test.php

```diff
@@ -178,23 +178,88 @@ public function testGetAchievementOfTheWeek(): void
             ]);
     }
 
-    // public function testGetAchievementsEarnedBetween(): void
-    // {
-    //     // TODO
-    //
-    //     $this->get($this->apiUrl('GetAchievementsEarnedBetween'))
-    //         ->assertSuccessful()
-    //         ->assertExactJson([]);
-    // }
-    //
-    // public function testGetAchievementsEarnedOnDay(): void
-    // {
-    //     // TODO
-    //
-    //     $this->get($this->apiUrl('GetAchievementsEarnedOnDay'))
-    //         ->assertSuccessful()
-    //         ->assertExactJson([]);
-    // }
+    public function testGetAchievementsEarnedBetweenEmptyResponse(): void
+    {
+        $this->get($this->apiUrl('GetAchievementsEarnedBetween'))
+            ->assertSuccessful()
+            ->assertExactJson([]);
+    }
+
+    public function testGetAchievementsEarnedBetween(): void
+    {
+        /** @var System $system */
+        $system = System::factory()->create();
+        /** @var Game $game */
+        $game = Game::factory()->create(['ConsoleID' => $system->ID]);
+        /** @var Achievement $achievement */
+        $achievement = Achievement::factory()->published()->create(['GameID' => $game->ID, 'Points' => 100]);
+        /** @var PlayerAchievement $unlock */
+        $unlock = PlayerAchievement::factory()->create(['AchievementID' => $achievement->ID, 'User' => $this->user->User]);
+
+        $this->get(
+            $this->apiUrl('GetAchievementsEarnedBetween', [
+                'u' => $this->user->User,
+                'f' => Carbon::now()->subDay()->startOfDay()->unix(),
+                't' => Carbon::now()->addDay()->endOfDay()->unix(),
+            ])
+        )
+            ->assertSuccessful()
+            ->assertJson([
+                [
+                    'AchievementID' => $achievement->ID,
+                    'ConsoleName' => $system->Name,
+                    'CumulScore' => 100,
+                    'Date' => $unlock->Date->format('Y-m-d H:i:s'),
+                    'Description' => $achievement->Description,
+                    'GameID' => $game->ID,
+                    'GameIcon' => '/Images/000001.png',
+                    'GameTitle' => $game->Title,
+                    'GameURL' => '/game/' . $game->ID,
+                    'HardcoreMode' => UnlockMode::Softcore,
```

Comment:
Additionally, `AchievementID`, `Points`, `GameID` were also previously returned as strings. Only `CumulScore` was returned as an integer.

Here's output comparing master to this branch:
```
[{"Date":"2018-03-31 18:23:56","HardcoreMode":"0","AchievementID":"4347",
"Title":"Mighty Mouse","Description":"Rescue Minnie From Mizrabel on Hard Difficulty",
"BadgeName":"04977","Points":"50","Author":"Scott",
"GameTitle":"Castle of Illusion Starring Mickey Mouse","GameIcon":"\/Images\/048628.png",
"GameID":"9","ConsoleName":"Mega Drive","CumulScore":50,"BadgeURL":"\/Badge\/04977.png",
"GameURL":"\/game\/9"},
{"Date":"2018-03-31 18:23:56","HardcoreMode":"1","AchievementID":"4347",
"Title":"Mighty Mouse","Description":"Rescue Minnie From Mizrabel on Hard Difficulty",
"BadgeName":"04977","Points":"50","Author":"Scott",
"GameTitle":"Castle of Illusion Starring Mickey Mouse","GameIcon":"\/Images\/048628.png",
"GameID":"9","ConsoleName":"Mega Drive","CumulScore":100,"BadgeURL":"\/Badge\/04977.png",
"GameURL":"\/game\/9"}]
```
```
[{"Date":"2018-03-31 18:23:56","HardcoreMode":0,"AchievementID":4347,
"Title":"Mighty Mouse","Description":"Rescue Minnie From Mizrabel on Hard Difficulty",
"BadgeName":"04977","Points":50,"Author":"Scott",
"GameTitle":"Castle of Illusion Starring Mickey Mouse","GameIcon":"\/Images\/048628.png",
"GameID":9,"ConsoleName":"Mega Drive","CumulScore":50,"BadgeURL":"\/Badge\/04977.png",
"GameURL":"\/game\/9"},
{"Date":"2018-03-31 18:23:56","HardcoreMode":1,"AchievementID":4347,
"Title":"Mighty Mouse","Description":"Rescue Minnie From Mizrabel on Hard Difficulty",
"BadgeName":"04977","Points":50,"Author":"Scott",
"GameTitle":"Castle of Illusion Starring Mickey Mouse","GameIcon":"\/Images\/048628.png",
"GameID":9,"ConsoleName":"Mega Drive","CumulScore":100,"BadgeURL":"\/Badge\/04977.png",
"GameURL":"\/game\/9"}]
```

Reply:
Laravel's database adpaters treat integer columns as actual integers thus making these integers in the response, too.

As discussed on Discord - depending on whether we consider moving the casts from strings to integers to be a breaking change I can make sure this change is reverted; however this is the only kind of change that I would've allowed as any API client would probably cast those to integers anyways.

What would you suggest? Keeping them or making sure the response stays the same?

Reply:
I think it's a reasonable change, but how many times have we made assumptions about what clients are actually doing that has bitten us?

Reply:
Once, technically, because of the removal of the user ID - that was an oversight on my part. The empty object vs. array response was never allowed to go through. Which reminds me that i should write a test for that, too.

Let's keep it clean, no assumptions, you're right. I'll add the string casts 👍

Reply:
Looking into it, adding the string casts would add some amount of complexity to the web api endpoints' scripts. Should we just allow strings being casted to ints as is?

Reply:
> Should we just allow strings being casted to ints as is?

Not sure I understand what you're asking. Changing the return type is less dangerous than removing or renaming a field. I was just noting the change expecting it to cause problems downstream. If you don't think that's something to be concerned about, or feel it's a desirable change regardless of the impact, I'm fine with it.

Reply:
Yeah, that's what I was meant; I don't think it's something to be concerned about, no.

In case we introduce a breaking change because of those strings turning into integers for any of our clients, for whatever reason, then I'll take care of fixing it.